File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    answer = 0
    # 출른 희망 시각 + 10분 계산
    for i in range(len(schedules)):
        schedules[i] = ((schedules[i] // 100) + (((schedules[i] % 100) + 10) // 60)) * 100 + (
                schedules[i] % 100 + 10) % 60
    for s in range(len(schedules)):
        check = False
        for i in range(7):
            currentDay = ((startday - 1) + i) % 7

            if currentDay != 5 and currentDay != 6:
                if timelogs[s][i] > schedules[s]:
                    check = not check
                    break
        if not check:
            answer += 1

    return answer

File: 프로그래머스/1/388351. 유연근무제/유연근무제.java
class Solution {
       public int solution(int[] schedules, int[][] timelogs, int startday) {
        int answer = 0;
        // 출근 희망 시간 + 10분 으로 업데이트하기
        for (int i = 0; i < schedules.length; i++) {
            int time = schedules[i];
            schedules[i] = (time / 100 + (((time % 100) + 10) / 60)) * 100 + ((time % 100) + 10) % 60;
        }

        for (int i = 0; i < schedules.length; i++) {
            boolean check = true;
            int dayCnt = 0;
            for (int j = 0; j < 7; j++) {
                int currentDay = (startday - 1 + j) % 7;
                if (currentDay < 5) {
                    if (timelogs[i][j] > schedules[i]) {
                        check = !check;
                        break;
                    }
                }
            }
            if (check) answer++;

        }
        return answer;
    }
}